app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreation.kt

```diff
@@ -32,14 +38,19 @@ class MeetUpCreation : AppCompatActivity() {
 
         dateFormat = SimpleDateFormat(getString(R.string.date_long_format))
 
-        if (intent.hasExtra(MEETUP_EDIT)) {
-            val meetup = intent.getSerializableExtra(MEETUP_EDIT) as MeetUp
-            fillFields(meetup)
-        } else {
-            model.startDate.value = Calendar.getInstance()
-            model.endDate.value = Calendar.getInstance()
+        loadIntent()
+
+        if (savedInstanceState == null) {
```

Comment:
Overall looks good to me, I believe that some part of the code are repeating multiple times, maybe this could be refactored in a "utils" to avoid code replication. I noticed this check and also the tagsViewModel (see other comment)